### ISSUE TITLE
Pinned Bookmarks are not saved in settings export.

### ISSUE THREAD

**Issue by Bamba-Tom** [Kiuryy/Bookmark_Sidebar#101]
I used Pinned Bookmarks as the source for the grid 'speed dials' when I was testing a pre-1.8.2 build. 
After updating to 1.8.2 and importing from a saved config they were gone, and can see they're not in any exported json backup.

**Comment by Kiuryy** [comment]
Pinned bookmarks are not being synchronised across your devices and are not included in the export json.

The reason for this is, that Chrome bookmarks does not have a unique identifier. So when you e.g. using a computer and have a bookmark pinned (which is done based on the bookmark ID), on the laptop the same bookmark might have another ID and therefore another bookmark will be pinned there (or none, if the ID does not exist on the laptop) ... or the IDs are the same and it's working as intended. It seems like Google does not want you to rely on this ID and only access the bookmarks through their API, which does not support saving additional information (like pinned=true, a custom icon, tags, ...). Adding the bookmark IDs in the export json would then only work reliable if you use it to import the file again on the same device with the same browser and profile.

These non-permanent IDs seem to only happen if you have bookmark sync enabled:
https://groups.google.com/a/chromium.org/forum/#!topic/chromium-extensions/TuUfU7HBLtY

That's why I could not (or to be precisely, didn't want to) add more bookmark related features to the sidebar with this restriction. Maybe they will extend the API to include an account wide unique ID in the future.

A possible workaround would be some sort of bookmark detection which is using the URL and title and the parent folder(s) to create a unique hash. But this is very unreliable when something at these data change (e.g. renaming of a bookmark), because the detection of these changes are not very reliable, since there are many ways to change bookmarks/folders (for instance when changing them not using the sidebar, but the default bookmark manager, ...)

**Comment by Bamba-Tom** [comment]
Thanks for the explanation. I posted my other feature request before reading your reply. This reason will apply to that too (re folder IDs) I expect?

**Comment by Kiuryy** [comment]
You are right, unfortunately, this problem applies to your other feature suggestion as well.

**Comment by hello-smile6** [comment]
What if the extension syncs the bookmarks, not the browser?
